lib/graphql/execution/execute.rb

```diff
@@ -164,14 +162,19 @@ def resolve_value(parent_type, field_defn, field_type, value, selection, field_c
             result
           when GraphQL::TypeKinds::NON_NULL
             wrapped_type = field_type.of_type
-            resolve_value(
+            inner_value = resolve_value(
               parent_type,
               field_defn,
               wrapped_type,
               value,
               selection,
               field_ctx,
             )
+            if inner_value.nil?
+              PROPAGATE_NULL
```

Comment:
When can this happen?  If I remove this `if` condition that you added the tests still pass.  Because it seems like if we did get here, then `field_ctx.schema.type_error` should be used to avoid ignoring the invalid null